test/test_quantized.py

```diff
@@ -66,6 +66,61 @@ def qlinear_ref(X_q, X_scale, X_zp, W_q, W_scale, W_zp, b_q, Y_scale, Y_zp):
     Y_q_ref = _quantize(Prod_XqWq_ref, Y_scale / (X_scale * W_scale), Y_zp)
     return Y_q_ref
 
+# Reference quantized conv operator
+def qconv_ref(X, X_scale, X_zp, Weight, W_scale, W_zp, bias, stride, pad, dilation, groups, Y_scale, Y_zp):
+    # NCHW for X
+    # KCRS for W
+    # result_ref is NCHW as well
+    assert X.dim() == 4, "activations should be a 4-D tensor"
+    assert Weight.dim() == 4, "weights should be a 4-D tensor"
+    N, IC, H, W = X.size()
+    OC, IC_per_G, kernel_h, kernel_w = Weight.size()
+
+    OH = int(_conv_output_shape(H, kernel_h, pad[0], stride[0], dilation[0]))
+    OW = int(_conv_output_shape(W, kernel_w, pad[1], stride[1], dilation[1]))
+    assert dilation[0] == 1 and dilation[1] == 1, "Dilation should be 1"
+    # perform conv and calculate row offsets as well
+    result_ref = torch.zeros([N, OC, OH, OW], dtype=torch.float)
+    row_offsets_ref = torch.zeros([N, OC, OH, OW], dtype=torch.float)
+    for i in range(N):
```

Comment:
The nested loops will make your test run long, and if you decide to use it in the hypothesis tests, it will almost certainly violate the deadline of 200ms. A better way is to use the `torch.nn.functional.conv2d` and do the quantization trick at the input/output

Reply:
Please use functional.conv2d, having zero-point map to float zero should work and give you close values, that you can compare with allclose.